### ISSUE TITLE
2.3.2 broke my frontmatter processing template

### ISSUE THREAD

**Issue by luckman212** [SilentVoid13/Templater#1380]
Not sure exactly _why_ yet, but [this commit](https://github.com/SilentVoid13/Templater/commit/263919674a4186632651a7f7b4296b9c3b047f6e) has broken [my template that processes frontmatter](https://github.com/SilentVoid13/Templater/issues/109#issuecomment-2122686242). It no longer does the replacement. No error in console, frontmatter just flickers and gets replaced with the save content as before. I had to revert to 2.3.1.

### Plugin information

-   OS: macOS
-   Templater version: 2.3.2
-   Obsidian version: 1.6.0 (Installer 1.5.12)
-   Template folder location: `templates/`
-   Timeout: `5`

**Comment by Zachatoo** [comment]
Instead of a timeout I would recommend seeing if the [hooks module](https://silentvoid13.github.io/Templater/internal-functions/internal-modules/hooks-module.html) resolves the issue for you. Try wrapping your editor changes in the `on_all_templates_executed` hook.

**Comment by luckman212** [comment]
Thanks @Zachatoo but I updated my code to use hooks, and still can't get it to work in 2.3.2. I updated the top post to point to a new version of the script that uses `tp.hooks.on_all_templates_executed()`

**Comment by tobyfb** [comment]
I might have a problem related to this. All my templates that use regex are built with the following structure / logic:

```
<%*
// based on the following example: https://github.com/SilentVoid13/Templater/discussions/255; relevant for other
var content = tp.file.content;
content = content.replace(/updated:(.*)/g, "updated: " + tp.date.now("YYYY-MM-DDTHH:mm:ss"));
var file = app.workspace.activeLeaf.view.file; // *get current file*
app.vault.modify(file, content); // *replace content with new content*
%>
```

This stopped working for me with 2.3.2.

I see a flicker where content is replaced as expected and then after a split second, things revert. It would be great to know if this is a bug or intended behavior and how I might need to adjust all of my templater templates for this to work.

**Comment by FeralFlora** [comment]
> frontmatter just flickers and gets replaced with the save content as before

I was able to reproduce a similar issue when using "Create new note from template" with a simple template that includes `tp.hooks.on_all_templates_executed` to add frontmatter: https://github.com/SilentVoid13/Templater/issues/1309#issuecomment-2124385751

After the template insertion fix in 2.3.2, template insertion works (surprise!), but the frontmatter disappears when using "create new note from template" with the same template.

**Comment by luckman212** [comment]
Not quite sure _how_, but I fixed my template and it's [working](https://github.com/SilentVoid13/Templater/issues/109#issuecomment-2128182167) with 2.3.2 now.

**Comment by Zachatoo** [comment]
Should be fixed in 2.3.3.
